starport/pkg/cosmosgen/templates/vuex/store.ts.tpl

```diff
@@ -0,0 +1,112 @@
+import { txClient, queryClient } from './module'
+
+{{ range .Module.Types }}import { {{ .Name }} } from "./module/types/{{ resolveFile .FilePath }}"
+{{ end }}
+
+async function initTxClient(vuexGetters) {
+	return await txClient(vuexGetters['chain/common/wallet/signer'], {
+		addr: vuexGetters['chain/common/env/apiTendermint']
+	})
+}
+
+async function initQueryClient(vuexGetters) {
+	return await queryClient({
+		addr: vuexGetters['chain/common/env/apiCosmos']
+	})
+}
+
+function getStructure(template) {
+	let structure = { fields: [] }
+	for (const [key, value] of Object.entries(template)) {
+		let field: any = {}
+		field.name = key
+		field.type = typeof value
+		structure.fields.push(field)
+	}
+	return structure
+}
+
+const getDefaultState = () => {
+	return {
+        {{ range .Module.Queries }}get{{ .Name }}: (state) => (params = {}) => {
+			return state.Post[JSON.stringify(params)] ?? {}
+		},
+        {{ end }}
+        _Structure: {
+            {{ range .Module.Types }}{{ .Name }}: getStructure({{ .Name }}.fromPartial({})),
+            {{ end }}
+		},
+		_Subscriptions: new Set(),
+	}
+}
+
+// initial state
+const state = getDefaultState()
+
+export default {
+	namespaced: true,
+	state,
+	mutations: {
+		RESET_STATE(state) {
+			Object.assign(state, getDefaultState())
+		},
+		QUERY(state, { query, key, value }) {
+			state[query][JSON.stringify(key)] = value
+		},
+		SUBSCRIBE(state, subscription) {
+			state._Subscriptions.add(subscription)
+		},
+		UNSUBSCRIBE(state, subscription) {
+			state._Subscriptions.delete(subscription)
+		}
+	},
+	getters: {
+        {{ range .Module.Queries }}get{{ .Name }}: (state) => (params = {}) => {
+			return state.Post[JSON.stringify(params)] ?? {}
+		},
+        {{ end }}
+		getTypeStructure: (state) => (type) => {
+			return state._Structure[type].fields
+		}
+	},
+	actions: {
+		init({ dispatch, rootGetters }) {
+			console.log('init')
+			if (rootGetters['chain/common/env/client']) {
+				rootGetters['chain/common/env/client'].on('newblock', () => {
+					dispatch('StoreUpdate')
+				})
+			}
+		},
+		resetState({ commit }) {
+			commit('RESET_STATE')
+		},
+		unsubscribe({ commit }, subscription) {
+			commit('UNSUBSCRIBE', subscription)
+		},
+		async StoreUpdate({ state, dispatch }) {
+			state._Subscriptions.forEach((subscription) => {
+				dispatch(subscription.action, subscription.payload)
+			})
+		},
+        {{ range .Module.Queries }}async {{ .FullName }}({ commit, rootGetters }, { subscribe = false, ...key }) {
+			try {
+				const value = (await (await initQueryClient(rootGetters)).{{ camelCase .FullName }}.apply(null, Object.values(key))).data
```

Comment:
@clockworkgr I used `x.apply(null, Object.values(key))` here instead of `x(keys.id)` to make it generic to all modules.

Reply:
although that does make it easier on code generation, key is an object ans as such guarantees on property ordering are somewhat iffy (although have been quite consistent since ES2015) but are still based on insertion order. Thus, It's likely to lead into issues as the frontend dev (for whatever reason) may populate the key params in an order unexpected by the Query function. 

In short....not confident about this one :/

Reply:
I'll be providing more metadata through `cosmosanaylsis/module` package about these props by parsing the http annotions inside proto files. So, we can use them for `getters` and here to define arguments.

Reply:
added new metadata here: https://github.com/tendermint/starport/blob/9c418bbd43a1077de5b8c39aa1d6ff0efa875858/starport/pkg/cosmosanalysis/module/module_test.go#L89-L91